### ISSUE TITLE
`new File()` inconsistent behavior with NodeJS

### ISSUE THREAD

**Issue by jasongitmail** [oven-sh/bun#13137]
### What version of Bun is running?

1.1.21+70ca2b76c

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

Start a node repl and a bun repl, and then:
```js
const file = new File([], "empty.txt", { type: "text/plain" })
file.name
file.size
file.type
```

Results:
<img width="1314" alt="file" src="https://github.com/user-attachments/assets/9f3aaced-e72c-4fef-8fff-e3c6a16cfb19">

### What is the expected behavior?

Expected to match NodeJS's behavior:

```js
const file = new File([], "empty.txt", { type: "text/plain" })
file.name // "empty.txt"
file.size // 0 
file.type // text/plain
```

### What do you see instead?

Actual output with Bun:

```js
const file = new File([], "empty.txt", { type: "text/plain" })
file.name // undefined
file.size // 0 
file.type // text/plain;charset=utf-8
```

- `file.name` is expected to be "empty.txt".
- `file.type` is expected to be "text/plain" without the chartset, to match Node.

### Additional information

Despite Bun's repl warning, this is not repl related.

I encountered this in my unit tests using `bun:test`. Then reproduced it in the repl to demonstrate it.

It's small, but "if it doesn't match Node it's a bug", so I wanted to report it.

**Comment by Electroid** [comment]
Can you try with `bun upgrade --canary`, pretty sure we fixed this bug recently but haven't released it yet.

**Comment by jasongitmail** [comment]
I upgraded to canary and received the same result in the repl. Still has the reported issue with name and type.

<img width="976" alt="bun" src="https://github.com/user-attachments/assets/a0d0de81-7bb6-4c0a-b730-abca0402f433">

**Comment by huseeiin** [comment]
this only happens with `[]` (empty) files

**Comment by jasongitmail** [comment]
yep, my other tests for non-empty files pass consistently between and node
